html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+
+Let's look at some keyframe animation syntax. Like transitions, there is a long and short form of using animations. We have used the long form here so you can get to grips with some of the main animation sub-properties that you will use to get started. 
+
+~~~css
+#ball {
+  /* ... other css properties ... */
+  animation-duration: 3s;
+  animation-name: bounce;
+  animation-iteration-count: infinite;
+  animation-direction: alternate;
+}
+
+@keyframes bounce {
+  from {
+    margin-top: 100%;
+  }
+
+  to {
+    margin-top: 0%;
+  }
+}
+~~~
+
+Note that this process has two distinct parts. The first part is the configuration stage where we define our animation sub-properties on the `#ball` element. Here we have an `animation-duration` of three seconds. This means that it will take three seconds for the `#ball` element to complete one animation cycle. Next we have defined the `animation-name` to be "bounce" which is essential for the `@keyframe` section coming up next. This is just a custom name that is not a particular CSS value. We could have called it "pineapples" if we so wished, but for our purposes "bounce" suits well. Then our `animation-iteration-count` is set to `infinite` which means this animation will run forever. You could set this to 1, 2, or as many iterations as you wish. Finally our `animation-direction` decides if our animation should alternate direction on the completion of one cycle, or reset to the start point and repeat itself. Here we set it to `alternate`, which means that the `#ball` will smoothly animate back along the path it came from in reverse as opposed to skipping straight back to its original point.  
+
+Now we will cover defining the animation sequence itself using the `@keyframes` at-rule. You have seen that the `@keyframes` definition references the "bounce" name we defined earlier. Then, we use the `from` and `to` properties to change the `margin-top` of `#ball` from 100% to 0%. It's important to know that keyframes use a percentage to indicate the times for an animation to take place and that the `from` and `to` statements are actually aliases for `0%` and `100%` respectively. You can read `from/0%` as meaning "at zero seconds" and `to/100%` as "at 3 seconds" according to our `animation-duration`. There is no hard and fast rule on whether or not you should use `from/to` or `0%/100%`. Just pick a style and be consistent with it. The `from/to` statement also defines one animation cycle. So if we were to set our `animation-iteration-count` to 2 then the ball would change its `margin-top` from 100% to 0%, then from 0% to 100%, and then the animation would stop. Be careful not to think of one cycle as a 'round trip'. 
```

Comment:
this is mildly confusing because in addition to explaining the from/to 100%/0% thing, you are also animating margin-top from 100% to 0%... i'd just change the stuff you're animating to avoid any misunderstanding here.